packages/gateway/src/features/registration/root-resolvers.ts

```diff
@@ -636,20 +639,79 @@ export const resolvers: GQLResolver = {
 
       return taskEntry.resource.id
     },
-    async confirmRegistration(_, { id }, { headers: authHeader }) {
+    async confirmRegistration(_, { id, details }, { headers: authHeader }) {     
+      const token = authHeader.Authorization.replace("Bearer ", "").trim();
+      const recordSpecificToken = await tokenExchangeHandler(
+        token,
+        authHeader,
+        id
+      )
+
+      authHeader.Authorization = `Bearer ${recordSpecificToken.access_token}`
+
       if (!inScope(authHeader, ['record.confirm-registration'])) {
-        throw new Error(
-          'User does not have a "record.confirm-registration" scope'
-        )
+        throw new Error('User does not have a Confirm Registration scope');
+      }
+
+      if (!hasRecordAccess(authHeader, id)) {
+        throw new Error('User does not have access to the record');
+      }
+
+      try {
+        const taskEntry = await confirmRegistration(
+          id,
+          authHeader,
+          {
+            error: details.error,
+            registrationNumber: details.registrationNumber,
+            childIdentifiers: details.identifiers,
+            compositionId: id
+          }
+        );
+
+        return taskEntry.resource.id;
+      } catch (error) {
+        throw new Error(`Failed to confirm registration: ${error.message}`);
+      }
+    },
+    async rejectRegistration(
+      _,
+      { id, details },
+      { headers: authHeader }
+    ) {
+
+      const token = authHeader.Authorization.replace("Bearer ", "").trim();
+      const recordSpecificToken = await tokenExchangeHandler(
+        token,
+        authHeader,
+        id
+      )
+
+      authHeader.Authorization = `Bearer ${recordSpecificToken.access_token}`
+
+      if (!inScope(authHeader, ['record.reject-registration'])) {
+        throw new Error('User does not have a Reject Registration" scope');
       }
 
       if (!hasRecordAccess(authHeader, id)) {
-        throw new Error('User does not have access to the record')
+        throw new Error('User does not have access to the record');
```

Comment:
Please check that you are using Prettier correctly - our config doesn't use semi-colons.